### ISSUE TITLE
I get the cider-nrepl warning, but I can't find its cause.

### ISSUE THREAD

**Issue by joostkremers** [clojure-emacs/cider#2421]
Hi,

Suddenly yesterday I started getting the cider-nrepl warning whenever I'm starting a new repl:

> WARNING: CIDER 0.18.0-snapshot requires cider-nrepl 0.18.0+, but you're currently using cider-nrepl 0.18.0. Things will break!

I followed the link provided, but there turned out not to be a file `profiles.clj` in `~/.lein/`, and `project.clj` doesn't mention `cider-nrepl` at all. (I use leiningen, not boot, so there's no `build.boot` or `~/.boot/profile.boot` to check) So that leaves me wondering where the problem lies.

I'm a Clojure newbie, and I just followed the getting started guides for installing Clojure and Cider and didn't configure anything special. (What I'm saying is that I don't really have an idea of how everything works, what components exist where and what they do. Heck, I can't even find the code for `cider/cider-nrepl` on my system...)

I create new projects with `lein new app <name>` and I get the warning even with a freshly created project. Things were fine for the past couple of weeks, this just started happening yesterday, as mentioned.

Version info:

> Elisp> (emacs-version)
> "GNU Emacs 26.1.50 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.18.9)\n of 2018-08-13"

> -;; CIDER 0.18.0snapshot (package: 20180817.153), nREPL 0.2.13
> ;; Clojure 1.8.0, Java 10.0.2

The `*Messages*` buffer says this after starting a repl:

> [nREPL] Starting server via /home/joost/bin/lein update-in :dependencies conj \[org.clojure/tools.nrepl\ \"0.2.13\"\ \:exclusions\ \[org.clojure/clojure\]\] -- update-in :plugins conj \[cider/cider-nrepl\ \"0.18.0\"\] -- repl :headless :host ::...

Which suggests to me that Cider is indeed trying to use cider-nrepl version 0.18.0, but I can't figure out whether I configured it to do so.

**Comment by joostkremers** [comment]
Actually, I think I already had the latest version installed, because `list-packages` didn't list an update for Cider. (The version info above seems to confirm this: it says `package 20180817.153`). Just in case, I deleted the package, restarted Emacs and reinstalled, but the error remains.

However, any chance that 

    (version<= middleware-version cider-required-middleware-version)

should actually be 

    (version< middleware-version cider-required-middleware-version)

?

I ran `cider--check-middleware-compatibility` through Edebug and both `middleware-version` and `cider-required-middleware-version` are `"0.18.0"` in my case, so regardless of the order of arguments, `version<=` is going to be true...
